src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -184,10 +186,13 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
   }
 
   def appendHeader(header: Header, votingSettings: VotingSettings): Try[ErgoStateContext] = {
+    if (lastHeaders.isEmpty) {
+      log.info("Last headers are empty!")
+    }
     validateVotes(header)
       .validate(hdrHeight,
-        lastHeaderOpt.map(_.height + 1).getOrElse(ErgoHistory.GenesisHeight) == header.height,
-        s"${header.id} => ${lastHeaderOpt.map(_.height)} == ${header.height}")
+        lastHeaders.isEmpty || lastHeaders.head.height + 1 == header.height,
+        s"${header.id} => ${lastHeaderOpt.map(_.height + 1)} == ${header.height}")
```

Comment:
these changes are suspicious, please provide explanation for the condition in the code.